lib/matplotlib/pyplot.py

```diff
@@ -2436,12 +2436,14 @@ def imsave(
 
 def matshow(A: ArrayLike, fignum: None | int = None, **kwargs) -> AxesImage:
     """
-    Display an array as a matrix in a new figure window.
-
-    The origin is set at the upper left hand corner and rows (first
-    dimension of the array) are displayed horizontally.  The aspect
-    ratio of the figure window is that of the array, unless this would
-    make an excessively short or narrow figure.
+    Display a 2D array as a matrix in a new figure window.
+
+    The origin is set at the upper left hand corner.
+    The indexing is `(row, column)`, where
+    the first index represents the height of the figure
+    and the second index represents the width of the figure.
```

Comment:
Sorry to bikeshed again, but IMHO this is logically not correct. The index has no direct relation to the height - it's only that first index and height are associated with the same direction (=vertical). Do you have any reservations on my original proposal:

```
    The indexing is ``(row, column)`` so that the first index runs vertically
    and the second index runs horizontally in the figure.
```